lib/publiccloud/aws_client.pm

```diff
@@ -0,0 +1,110 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for amazon connection and authentication
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>, qa-c team <qa-c@suse.de>
+
+package publiccloud::aws_client;
+use Mojo::Base -base;
+# use Mojo::JSON 'decode_json';
+use testapi;
+# use mmapi 'get_current_job_id';
+
+use constant CREDENTIALS_FILE => '/root/amazon_credentials';
+
+has key_id => undef;
+has key_secret => undef;
+has security_token => undef;
+has region => undef;
+has ssh_key => undef;
+has ssh_key_file => undef;
+has credentials => undef;
+has vault => undef;
+has aws_account_id => undef;
+has service => undef;
+
+sub vault_create_credentials {
+    my ($self) = @_;
+    my $data = undef;
+
+    if ($self->service =~ /ECR|EC2/) {
+        record_info('INFO', 'Get credentials from VAULT server.');
+        $data = $self->vault->get_secrets('/aws/creds/openqa-role');
+    }
+    elsif ($self->service eq "EKS") {
+        record_info('INFO', 'Get credentials from VAULT server for EKS');
+        my $path = '/aws/sts/openqa-role-eks';
+        my $res
+          = $self->vault->api('/v1/' . get_required_var('PUBLIC_CLOUD_VAULT_NAMESPACE') . $path, method => 'post');
+        $data = $res->{data};
+        $self->security_token($data->{security_token});
+        die('Failed to retrieve token') unless (defined($self->security_token));
+    }
+    else {
+        die('Invalid service: ' . $self->service);
+    }
+
+    die('Failed to retrieve key') unless (defined($data->{access_key}) && defined($data->{secret_key}));
+    $self->key_id($data->{access_key});
+    $self->key_secret($data->{secret_key});
+}
+
+sub _check_credentials {
+    my ($self) = @_;
+    if ($self->service =~ /ECR|EC2/) {
+        my $max_tries = 6;
+        for my $i (1 .. $max_tries) {
+            my $out = script_output('aws ec2 describe-images --dry-run', 300, proceed_on_failure => 1);
+            return 1 if ($out !~ /AuthFailure/m && $out !~ /"aws configure"/m);
+            sleep 30;
+        }
+    }
+    elsif ($self->service eq "EKS") {
+        return 1;
```

Comment:
Don't we need checks as well on EKS?

Reply:
I think we do , but maybe it can be done in separate PR because this is just code refactoring and this skip of the check is already there ...

Reply:
Yes, I agree. In a next PR. I created a ticket https://progress.opensuse.org/issues/102774

Reply:
Perfect!